### ISSUE TITLE
Stack overflow mapping anonymous type

### ISSUE THREAD

**Issue by kwaclaw** [agileobjects/AgileMapper#9]
Given these classes:
```
    [CLSCompliant(true)]
    public abstract class ExportFormatOptions: ICloneable
    {
        protected ExportFormatOptions();

        public virtual object Clone();
    }

    [XmlRoot]
    public class TextFormatOptions: ExportFormatOptions
    {
        public TextFormatOptions();
        public TextFormatOptions(TextFormatOptions formatOption);

        [XmlAttribute]
        public long LinesPerPage { get; set; }
        [XmlAttribute]
        public long CharactersPerInch { get; set; }

        public override object Clone();
    }
```

I get a stack overflow when mapping an anonymous type to a new instance, but not when updating an existing instance (code commented out for the latter):

```
            var exportFormatOptions = new
            {
                CharactersPerInch = 16,
                LinesPerPage = 0
            };
            var target = Mapper.Map(exportFormatOptions).ToANew<TextFormatOptions>();
            //var target = new TextFormatOptions();
            //Mapper.Map(exportFormatOptions).Over(target);
```

**Comment by SteveWilkes** [comment]
I'll look at this properly soon, but I'm guessing the Mapper is picking the TextFormatOptions(TextFormatOptions) constructor and getting itself into an infinite loop that way.

**Comment by SteveWilkes** [comment]
I've committed [a fix] for this - the mapper now ignores copy constructors when figuring out how to create an object.

**Comment by SteveWilkes** [comment]
Fixed in v0.10.1 - thanks for the bug report! :)
